src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -460,6 +460,57 @@ class HttpGoogleServicesDAO(
     }
   }
 
+  def getBilling(credential: Credential) = {
+    new Cloudbilling.Builder(httpTransport, jsonFactory, credential).build()
+  }
+
+  /**
+    * NOTE: This function will returns "false" in both of the following cases:
+    * if you don't have sufficient scopes
+    *   - Google's JSON response body will contain "message": "Request had insufficient authentication scopes."
+    * if you're not authorized to see the billing account
+    *   - Google's JSON response body will contain "message" : "The caller does not have permission"
+    */
+  protected def credentialOwnsBillingAccount(credential: Credential, billingAccountName: String): Future[Boolean] = {
+    val fetcher = getBilling(credential).billingAccounts().get(billingAccountName)
+    retryWithRecoverWhen500orGoogleError(() => {
+      blocking {
```

Comment:
can you use some async/callback pattern here, like map{_ => true}, instead of blocking? (same below)